### ISSUE TITLE
[Bug] Failure when upgrading from ElasticSearch/Kibana OSS 7.9.1 to OpenSearch OSS 2.4.1

### ISSUE THREAD

**Issue by jakedern-msft** [opensearch-project/OpenSearch-Dashboards#3817]
**Describe the bug**

I am unsure if this is something I'm doing or an actual bug with the product however I am unable to upgrade from ElasticSearch/Kibana OSS v7.9.1 to OpenSearch OSS 2.4.1 due to an error in migrating the `.kibana` index. I've made several attempts with fresh installations each time, but no luck.

**To Reproduce**
1. Deploy ElasticSearch and Kibana OSS v7.9.1 (single node each)
2. Create index and produce some logs to it
3. Stop ElasticSearch and Kibana
4. Copy contents of `nodes/*` to a new directory
5. Start OpenSeach 2.4.1 (pointed at same directory) and wait for it to be stable 
6. Start OpenSearch-Dashboards which begins the upgrade process
7. OpenSearch-Dashboards gets locked up with some communication error with OpenSearch, but does not provide details on the issue even with verbose logging turned on.

```
{"type":"log","@timestamp":"2023-04-10T18:29:38Z","tags":["debug","config"],"pid":669,"message":"Marking config path as handled: opensearchDashboards"}
{"type":"log","@timestamp":"2023-04-10T18:29:38Z","tags":["info","savedobjects-service"],"pid":669,"message":"Waiting until all OpenSearch nodes are compatible with OpenSearch Dashboards before starting saved objects migrations..."}
{"type":"log","@timestamp":"2023-04-10T18:29:38Z","tags":["info","savedobjects-service"],"pid":669,"message":"Starting saved objects migrations"}
{"type":"log","@timestamp":"2023-04-10T18:29:38Z","tags":["info","savedobjects-service"],"pid":669,"message":"Detected mapping change in \"properties.originId\""}
{"type":"log","@timestamp":"2023-04-10T18:29:38Z","tags":["info","savedobjects-service"],"pid":669,"message":"Creating index .kibana_2."}
{"type":"log","@timestamp":"2023-04-10T18:29:38Z","tags":["info","savedobjects-service"],"pid":669,"message":"Migrating .kibana_1 saved objects to .kibana_2"}
{"type":"log","@timestamp":"2023-04-10T18:29:38Z","tags":["debug","savedobjects-service"],"pid":669,"message":"Migrating saved objects index-pattern:mssql-log-index, config:7.9.1-SNAPSHOT, telemetry:telemetry"}
{"type":"log","@timestamp":"2023-04-10T18:29:38Z","tags":["error","opensearch","data"],"pid":669,"message":"[ResponseError]: Response Error"}
{"type":"log","@timestamp":"2023-04-10T18:29:38Z","tags":["warning","savedobjects-service"],"pid":669,"message":"Unable to connect to OpenSearch. Error: Response Error"}
{"type":"log","@timestamp":"2023-04-10T18:29:39Z","tags":["debug","status"],"pid":669,"status":{"level":"unavailable","summary":"[48] services are unavailable","detail": <truncated>}}
{"type":"log","@timestamp":"2023-04-10T18:29:39Z","tags":["debug","http"],"pid":669,"message":"OpenSearch Dashboards server is not ready yet get:http://127.0.0.1:5601/api/status."}
{"type":"log","@timestamp":"2023-04-10T18:29:41Z","tags":["error","opensearch","data"],"pid":669,"message":"[ResponseError]: Response Error"}
{"type":"log","@timestamp":"2023-04-10T18:29:43Z","tags":["debug","metrics"],"pid":669,"message":"Refreshing metrics"}
{"type":"log","@timestamp":"2023-04-10T18:29:43Z","tags":["error","opensearch","data"],"pid":669,"message":"[ResponseError]: Response Error"}
<repeats>
```

**Expected behavior**

**OpenSearch Version**
2.4.1

**Dashboards Version**
2.4.1

**Plugins**
Security plugin disabled, otherwise whatever is default. Here is a dump of `src/plugins`

```bash
$ ls src/plugins/
advanced_settings  data_source             index_pattern_management  newsfeed                                region_map                    tile_map            vis_type_markdown    vis_type_vega
apm_oss            data_source_management  input_control_vis         opensearch_dashboards_legacy            saved_objects                 timeline            vis_type_metric      vis_type_vislib
bfetch             dev_tools               inspector                 opensearch_dashboards_overview          saved_objects_management      ui_actions          vis_type_table       vis_type_xy
charts             discover                legacy_export             opensearch_dashboards_react             share                         url_forwarding      vis_type_table_new   visualizations
console            embeddable              management                opensearch_dashboards_usage_collection  telemetry                     usage_collection    vis_type_tagcloud    visualize
dashboard          expressions             maps_legacy               opensearch_dashboards_utils             telemetry_collection_manager  vis_builder         vis_type_timeline
data               home                    navigation                opensearch_ui_shared                    telemetry_management_section  vis_default_editor  vis_type_timeseries
```
** I've zipped all additional files mentioned below and attached them as opensearch-debug-info.zip**

**Pre-upgrade debug information (step 2)**

```bash
$ curl localhost:9200/_cat/indices
yellow open logstash-2023.04.10 e69F7UEdSUaveBvTT2gaEA 1 1 5348 0  1.7mb  1.7mb
green  open .kibana_1           gCvCI0LURqSqP1bkOU13hA 1 0    3 1 20.3kb 20.3kb

$ curl localhost:9200/_cat/aliases
.kibana .kibana_1 - - - -

$ curl localhost:9200/.kibana
<attached as pre-upgrade-kibana-index.json>

$ curl localhost:9200/.logstash-2023.04.10
<attached as pre-upgrade-logs-index.json>
```

Additional Files
* pre-upgrade-elasticsearch.log
* pre-upgrade-kibana.log
* pre-upgrade-elasticsearch.yaml
* pre-upgrade-kibana.yaml

**Post upgrade debug information (Steps 6/7)**

```bash
$ curl localhost:9200/_cat/indices
yellow open logstash-2023.04.10 e69F7UEdSUaveBvTT2gaEA 1 1 12329 0  2.7mb  2.7mb
green  open .kibana_2           ThJTIoijQUWkbfhWpzQKRg 1 0     0 0   208b   208b
green  open .kibana_1           gCvCI0LURqSqP1bkOU13hA 1 0     3 0 20.7kb 20.7kb

$ curl localhost:9200/_cat/aliases
.kibana .kibana_1 - - - -

$ curl localhost:9200/.kibana_1
<attached as post-upgrade-opensearch-kibana-index.json>

$ curl localhost:9200/.kibana_2
<attached as post-upgrade-opensearch-kibana-2-index.json>

$ curl localhost:9200/logstash-2023.04.10
<attached as post-upgrade-opensearch-logs-index.json>
```

Additional Files
* post-upgrade-opensearch.log
* post-upgrade-opensearch-dashboards.log
* post-upgrade-opensearch.yaml
* post-upgrade-opensearch-dashboards.yaml

**Host/Environment (please complete the following information):**
 - Ubuntu 20.04 docker container running on Ubuntu 20.04 host

**Additional context**

Any help would be greatly appreciated!
[opensearch_debug_info.zip](https://github.com/opensearch-project/OpenSearch-Dashboards/files/11193181/opensearch_debug_info.zip)

**Comment by jakedern-msft** [comment]
Hey @aoguan1990 and @kavilla, I know investigating this is probably pretty low priority but do you have any pointers on at least getting more logging output from opensearch-dashboards? 

I want to do more investigation here, but am kind of stuck since even with `logging.verbose` turned on there are no details being provided as far as what error is occurring when trying to communicate with opensearch. `{"type":"log","@timestamp":"2023-04-10T18:29:43Z","tags":["error","opensearch","data"],"pid":669,"message":"[ResponseError]: Response Error"}` is not much to go on

**Comment by jakedern-msft** [comment]
@kavilla This seems like a more serious issue as we are hitting the same thing when upgrading from OpenSearch 2.4.1 to 2.7.0. Can you assist in looking into this?

**Comment by jakedern-msft** [comment]
I should add that the logs are identical to the ones that I have in the zip file in the original issue, here is some fresh output. OpenSearch is again reachable from the container:

```
$ curl http://127.0.0.1:9200
{
  "name" : "logsdb-0",
  "cluster_name" : "docker-cluster",
  "cluster_uuid" : "S3IjpUIWRqe_hgjRoK4kpA",
  "version" : {
    "distribution" : "opensearch",
    "number" : "2.7.0-SNAPSHOT",
    "build_type" : "tar",
    "build_hash" : "b7a6e09e492b1e965d827525f7863b366ef0e304",
    "build_date" : "2023-05-23T14:46:51.818283Z",
    "build_snapshot" : true,
    "lucene_version" : "9.5.0",
    "minimum_wire_compatibility_version" : "7.10.0",
    "minimum_index_compatibility_version" : "7.0.0"
  },
  "tagline" : "The OpenSearch Project: https://opensearch.org/"
}
```

This output is from withing the OpenSearch-Dashboards container. The logs once again indicate it can communicate with logsdb to create a new index, but then gets hung after trying to start the migration: 

```
{"type":"log","@timestamp":"2023-05-25T19:43:10Z","tags":["info","savedobjects-service"],"pid":677,"message":"Starting saved objects migrations"}
{"type":"log","@timestamp":"2023-05-25T19:43:10Z","tags":["info","savedobjects-service"],"pid":677,"message":"Detected mapping change in \"properties.visualization-visbuilder\""}
{"type":"log","@timestamp":"2023-05-25T19:43:10Z","tags":["info","savedobjects-service"],"pid":677,"message":"Creating index .opensearch_dashboards_2."}
{"type":"log","@timestamp":"2023-05-25T19:43:11Z","tags":["info","savedobjects-service"],"pid":677,"message":"Migrating .opensearch_dashboards_1 saved objects to .opensearch_dashboards_2"}
{"type":"log","@timestamp":"2023-05-25T19:43:11Z","tags":["error","opensearch","data"],"pid":677,"message":"[ResponseError]: Response Error"}
{"type":"log","@timestamp":"2023-05-25T19:43:11Z","tags":["warning","savedobjects-service"],"pid":677,"message":"Unable to connect to OpenSearch. Error: Response Error"}
{"type":"log","@timestamp":"2023-05-25T19:43:13Z","tags":["error","opensearch","data"],"pid":677,"message":"[ResponseError]: Response Error"}
{"type":"log","@timestamp":"2023-05-25T19:43:16Z","tags":["error","opensearch","data"],"pid":677,"message":"[ResponseError]: Response Error"}
{"type":"log","@timestamp":"2023-05-25T19:43:18Z","tags":["error","opensearch","data"],"pid":677,"message":"[ResponseError]: Response Error"}
{"type":"log","@timestamp":"2023-05-25T19:43:21Z","tags":["error","opensearch","data"],"pid":677,"message":"[ResponseError]: Response Error"}
```

Here is output from a curl command on the OpenSearch-Dashboards container showing the index was created successfully:

```
I have no name!@logsui-kcrcc:/var/log/opensearch-dashboards$ curl 127.0.0.1:9200/_cat/indices
green  open .opensearch_dashboards_2 RUAaC8eeRY6iXROijzdfag 1 0     0 0  208b  208b
green  open .opensearch_dashboards_1 Eqb5SsCOQPi21gKswHHKOQ 1 0     2 0 6.3kb 6.3kb
yellow open logstash-2023.05.25      vC998j1iTzmlw3kaESPD2A 1 1 11995 0 3.4mb 3.4mb
```

**Comment by ict-one-nl** [comment]
Ran into this as well when upgrading van 2.8.0 to 2.9.0 probably caused by cluster state not being green (because of increasing number of replicas in the primary cluster but not having enough nodes for that in the remote replication cluster)

**Comment by seraphjiang** [comment]
Thank you for reporting this issue. This issue has been open for over two years without recent activity, and multiple OpenSearch Dashboards releases have shipped since then with significant improvements and fixes.

We're closing this as part of a housekeeping effort to keep the backlog actionable. If this issue is still reproducible on a recent version of OpenSearch Dashboards (2.19+), please feel free to reopen this issue or file a new one with updated reproduction steps and version details.

We appreciate your contribution to OpenSearch!
